public/js/extended-access.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This client-side js file to initiate and manage
+ * Extended Access registration and entitlements
+ */
+
+import { insertHighlightedJson } from './utils.js';
+
+function revokeIDToken() {
+  google.accounts.id.revoke(readCookie('jwtSub'), done => {
+    console.log(done.error);
+  });
+  eraseCookie('jwtSub');
+}
+
+function createCookie(name, value, days=15) {
```

Comment:
We recommend using HTTP only cookies due to browser rules. Could we add a TODO to upgrade to these? 

[Documentation for Node](https://nodejs.org/dist/latest-v14.x/docs/api/http.html#http_response_setheader_name_value:~:text=or-,response.setHeader(%27Set%2DCookie%27%2C%20%5B%27type%3Dninja%27%2C%20%27language%3Djavascript%27%5D)%3B,-Attempting%20to%20set)